Backend/Microservices/GameService/src/Game.WebAPI/Middlewares/ExceptionHandlerMiddleware.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Net;
+using System.Text.Json;
+using FluentValidation;
+using Microsoft.EntityFrameworkCore;
+using OpenQA.Selenium;
+
+
+namespace Game.WebAPI.NewFolder
+{
+    public class ExceptionHandlerMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        public ExceptionHandlerMiddleware(RequestDelegate next) =>
+            _next = next;
+
+        public async Task Invoke(HttpContext context)
+        {
+            try
+            {
+                await _next(context);
+            }
+            catch (Exception exception)
+            {
+                await HandleExceptionAsync(context, exception);
+            }
+        }
+
+        private Task HandleExceptionAsync(HttpContext context, Exception exception)
+        {
+            var code = HttpStatusCode.InternalServerError;
+            var result = string.Empty;
+
+            switch (exception)
```

Comment:
You don't handle **InvalidOperationException** from `FirstAsync()` method in repositories. Conside use `FirstOrDefaultAsync()`